Library/include/CSP/Systems/Assets/GLTFMaterial.h

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "Material.h"
+
+namespace csp::systems
+{
+class GLTFMaterial;
+}
+
+namespace csp::json
+{
+class JsonSerializer;
+class JsonDeserializer;
+} // namespace csp::json
+
+void ToJson(csp::json::JsonSerializer& Serializer, const csp::systems::GLTFMaterial& Obj);
+void FromJson(const csp::json::JsonDeserializer& Deserializer, csp::systems::GLTFMaterial& Obj);
+
+namespace csp::systems
+{
+
+/// @brief Defines how to alpha value is interpreted
+/// The alpha value is taken from the fourth component of the base color for metallic-roughness material model
+enum class EAlphaMode
+{
+	Opaque,
+	Mask,
+	Blend
+};
+
+/// @ingroup Asset System
+/// @brief Data class which represents a GLTF material.
+class CSP_API GLTFMaterial : public Material
```

Comment:
Could be anything, these would be separate shaders. for example a fog card material that only appears when looking straight at it and smooth fades when close, or a spinny portal effect. 

Essentially because we cannot have runtime shaders, we will have to create a library of built in ones.  I'm keen to ensure adding these later is painless

Reply:
In that case I wonder if 'GLTFMaterial' the right name for the default standard material?

Reply:
This name is intentional so thats its clear to implementors that they should be using their existing gltf shaders to implement it.  Any questions on how to render this shader can then defer to the gltf material spec rather than some proprietary one